cumulus/test/service/src/lib.rs

```diff
@@ -0,0 +1,627 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Crate used for testing with Cirrus.
+
+#![warn(missing_docs)]
+
+pub mod chain_spec;
+
+use std::{future::Future, sync::Arc};
+
+use sc_basic_authorship::ProposerFactory;
+use sc_client_api::execution_extensions::ExecutionStrategies;
+use sc_network::{config::TransportConfig, multiaddr, NetworkService};
+use sc_service::{
+	config::{
+		DatabaseSource, KeepBlocks, KeystoreConfig, MultiaddrWithPeerId, NetworkConfiguration,
+		OffchainWorkerConfig, PruningMode, TransactionStorageMode, WasmExecutionMethod,
+	},
+	BasePath, ChainSpec, Configuration, Error as ServiceError, PartialComponents, Role,
+	RpcHandlers, TFullBackend, TFullClient, TaskManager,
+};
+use sp_arithmetic::traits::SaturatedConversion;
+use sp_blockchain::HeaderBackend;
+use sp_core::{Pair, H256};
+use sp_keyring::Sr25519Keyring;
+use sp_runtime::{codec::Encode, generic, traits::BlakeTwo256};
+use sp_trie::PrefixedMemoryDB;
+use substrate_test_client::{
+	BlockchainEventsExt, RpcHandlersExt, RpcTransactionError, RpcTransactionOutput,
+};
+
+use cirrus_client_service::{prepare_node_config, start_executor, StartExecutorParams};
+use cirrus_node_primitives::CollatorPair;
+use cirrus_test_runtime::{opaque::Block, Hash, RuntimeApi};
+use cumulus_client_consensus_relay_chain::PrimaryChainConsensus;
+
+pub use cirrus_test_runtime as runtime;
+pub use sp_keyring::Sr25519Keyring as Keyring;
+
+/// The signature of the announce block fn.
+pub type AnnounceBlockFn = Arc<dyn Fn(Hash, Option<Vec<u8>>) + Send + Sync>;
```

Comment:
Maybe `WrapAnnounceBlockFn` then?

Reply:
Hmm, no opinion on this, it was actually borrowed from cumulus-test-service.

Reply:
Well, I understand, but everywhere it is used variable is called wrap-something. So let's make it clearer.

Reply:
Updated.